### PR TITLE
[IMP] website_event: added two primary buttons (for outlook & gmail)

### PR DESCRIPTION
**Pull request by mba-odoo**
Task : https://www.odoo.com/web#id=39315&view_type=form&model=project.task&action=333&active_id=131&menu_id=
Pad : https://pad.odoo.com/p/r.4dce07daab55c55cfb3c26bc312dfefc

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by mba-odoo** [comment]
LGTM

File: addons/website_event/controllers/main.py [hunk 1]
```diff
@@ -257,8 +259,23 @@ def registration_confirm(self, event, **post):
...
+        urls = event._get_urls(Attendees.ids)
         return request.render("website_event.registration_complete", {
             'attendees': Attendees,
             'event': event,
+            'google_url': urls.get('google_url'),
+            'iCal_url': urls.get('iCal_url')
         })
+
+    @http.route(['/event/<model("event.event"):event>/ics/<string:name>.ics'], type='http', auth="public", website=True)
+    def calendar_appointment_ics(self, event, **kwargs):
```
Comment (tde-banana-odoo) [inline addons/website_event/controllers/main.py]: What is the purpose of name ? Why don't you define attendees from kwargs ? What happen if it is not set ?

+ update method and route naming

File: addons/website_event/controllers/main.py [hunk 2]
```diff
@@ -257,8 +259,23 @@ def registration_confirm(self, event, **post):
...
+        if not event or not event.registration_ids:
+            return request.not_found()
+        attendee_ids = list(literal_eval(kwargs.get('attendees')).values())
+        files = event.with_context(attendee_ids=attendee_ids).get_ics_file()
```
Comment (tde-banana-odoo) [inline addons/website_event/controllers/main.py]: Do not use context to propagate data.

File: addons/website_event/models/event.py [hunk 3]
```diff
@@ -115,3 +134,50 @@ def action_open_badge_editor(self):
...
+                raise UserError(_("First you have to specify the date of the invitation."))
+            cal_event.add('created').value = fields.Datetime.from_string(fields.Datetime.now()).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('dtstart').value = fields.Datetime.from_string(event.date_begin).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('dtend').value = fields.Datetime.from_string(event.date_end).replace(tzinfo=pytz.timezone('UTC'))
+            cal_event.add('summary').value = event.name
+            if event.address_id:
+                cal_event.add('location').value = event.sudo().address_id.contact_address
+
+            attendee_ids = self.env.context.get('attendee_ids')
+            attendees = self.env['event.registration'].sudo().browse(attendee_ids)
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event.py]: This is a security hole. Basically you allow people to browse whatever attendees in sudo using context.

File: addons/website_event/models/event.py [hunk 4]
```diff
@@ -115,3 +134,50 @@ def action_open_badge_editor(self):
...
+            for attendee in attendees:
+                attendee_add = cal_event.add('attendee')
+                attendee_add.value = u'MAILTO:' + (attendee.email or u'')
+            result[event.id] = cal.serialize().encode('utf-8')
+        return result
+
+    def _get_urls(self, attendees):
```
Comment (tde-banana-odoo) [inline addons/website_event/models/event.py]: Please use meaningful naming.

**Pull request merged**